services/statestorage/service.go

```diff
@@ -55,12 +56,18 @@ func (s *service) CommitStateDiff(input *services.CommitStateDiffInput) (*servic
 
 	s.reporting.Info("trying to commit state diff", log.BlockHeight(commitBlockHeight), log.Int("number-of-state-diffs", len(input.ContractStateDiffs)))
 
-	if lastCommittedBlock := s.lastCommittedBlockHeader.BlockHeight(); lastCommittedBlock+1 != commitBlockHeight {
+	lastCommittedBlock := s.lastCommittedBlockHeader.BlockHeight()
```

Comment:
this was originally inline and was broken only to facilitate debugging. normally i like to keep things easy to debug but we have also tend to prefer inline where a variable would not be used more than once... what do you think  (?)